src/tests/runner/test_runner.cpp

```diff
@@ -223,6 +225,15 @@ std::vector<Test::Result> run_a_test(const std::string& test_name)
       else if(std::unique_ptr<Test> test = Test::get_test(test_name))
          {
          std::vector<Test::Result> test_results = test->run();
+         for (auto& result : test_results)
+            {
+            if(test->registration_location() && !result.code_location())
```

Comment:
I would switch the checks as this fits better with the logic we want then reading from left to right.